src/lib/layouts/workflow-history-layout.svelte

```diff
@@ -71,18 +88,73 @@
     {#if workflow?.callbacks?.length}
       <WorkflowCallbacks callbacks={workflow.callbacks} />
     {/if}
-    <div
-      class="flex flex-col items-center gap-2 py-2 xl:flex-row xl:justify-between xl:gap-8"
-    >
-      <h2>
-        {translate('workflows.event-history')}
-      </h2>
-    </div>
   </div>
 </div>
-<div class="px-2 pb-24 md:px-4 lg:px-8">
-  <div class="flex w-full flex-col border border-subtle">
-    <TimelineGraph {workflow} {groups} {workflowTaskFailedError} />
-    <EventSummary {groups} {history} />
+<div class="relative px-2 pb-24 md:px-4 lg:px-8">
+  <div
+    class="flex items-center justify-between gap-2 py-2 xl:gap-8"
+    class:sticky-header={!$minimizeEventView}
+  >
+    <h2>
+      {translate('workflows.event-history')}
+    </h2>
+    <div class="flex items-center gap-2">
+      <ToggleButtons>
+        {#if $eventViewType !== 'json'}
+          <ToggleButton
+            icon={reverseSort ? 'descending' : 'ascending'}
+            data-testid="zoom-in"
+            on:click={onSort}
+            >{reverseSort ? 'Descending' : 'Ascending'}</ToggleButton
+          >
+        {/if}
+        <ToggleButton
+          icon={$minimizeEventView ? 'minimize' : 'expand'}
+          data-testid="expandAll"
+          on:click={() => ($minimizeEventView = !$minimizeEventView)}
+          >{$minimizeEventView ? 'Minimized' : 'Expanded'}</ToggleButton
+        >
+        <EventTypeFilter {compact} minimized={$minimizeEventView} />
+        <ToggleButton
+          disabled={!workflow.isRunning}
+          icon={$pauseLiveUpdates ? 'play' : 'pause'}
+          data-testid="pause"
+          on:click={() => ($pauseLiveUpdates = !$pauseLiveUpdates)}
+        >
+          {$pauseLiveUpdates ? 'Resume' : 'Pause'}
+        </ToggleButton>
+        <ToggleButton
+          data-testid="download"
+          icon="download"
+          on:click={() => (showDownloadPrompt = true)}
+        >
+          {translate('common.download')}
+        </ToggleButton>
+      </ToggleButtons>
+    </div>
+  </div>
+  <div
+    class="flex w-full flex-col border border-subtle {!$minimizeEventView &&
+      'border-t-0'}"
+  >
+    <TimelineGraph
+      {workflow}
+      {groups}
+      {workflowTaskFailedError}
+      viewportHeight={$minimizeEventView ? 360 : undefined}
+    />
+    <EventSummary {groups} {history} minimized={$minimizeEventView} />
   </div>
 </div>
+<DownloadEventHistoryModal
+  bind:open={showDownloadPrompt}
+  {namespace}
+  workflowId={workflow.id}
+  runId={workflow.runId}
+/>
+
+<style lang="postcss">
+  .sticky-header {
+    @apply sticky top-12 z-50 border-b border-subtle backdrop-blur-sm;
```

Comment:
This is an interesting styling choice 😂 
<img width="613" alt="Screenshot 2025-03-17 at 9 50 03 AM" src="https://github.com/user-attachments/assets/4cebfcbe-a4af-4267-ab61-38ce7f27aedc" />

Why not just a solid bg?

Reply:
bg-off-white/50 👍